### PR TITLE
TST: Add Python 3.7 support for Travis CI

### LINKED ISSUE #354: Add Python 3.7 in CI files
We should probably add support for Python 3.7 in the template!

### PR DESCRIPTION
**Pull request by pllim**
Fix #354

Like this, @astrofrog ?

### CODE REVIEW

File: {{ cookiecutter.package_name }}/.travis.yml [hunk 1]
```diff
@@ -66,11 +66,14 @@ env:
         - PYTHON_VERSION=2.7 SETUP_CMD='egg_info'
         - PYTHON_VERSION=3.5 SETUP_CMD='egg_info'
         - PYTHON_VERSION=3.6 SETUP_CMD='egg_info'
+        - PYTHON_VERSION=3.7 SETUP_CMD='egg_info'
         {%- elif  cookiecutter.minimum_python_version == "3.5" %}
         - PYTHON_VERSION=3.5 SETUP_CMD='egg_info'
         - PYTHON_VERSION=3.6 SETUP_CMD='egg_info'
+        - PYTHON_VERSION=3.7 SETUP_CMD='egg_info'
```
Comment (astrofrog) [inline {{ cookiecutter.package_name }}/.travis.yml]: For egg_info builds I don't think we need more than one Python 3 version, so can you remove 3.5 and 3.6 in the two cases above?

**Comment by pllim** [comment]
OK done. I am not sure why some lines have `{% if ...}` and some have `{%- if ...}` (with extra minus sign), please double check the syntax. Thanks.

File: {{ cookiecutter.package_name }}/.travis.yml [hunk 2]
```diff
@@ -95,7 +91,7 @@ matrix:
 
         # Now try Astropy dev with the latest Python and LTS with Python 2.7 and 3.x.
         - os: linux
-          env: ASTROPY_VERSION=development
+          env: PYTHON_VERSION=3.7 ASTROPY_VERSION=development
```
Comment (bsipocz) [inline {{ cookiecutter.package_name }}/.travis.yml]: I would rather like python 3.7 to be the default in L30 above? or is that still premature?

File: {{ cookiecutter.package_name }}/.travis.yml (hunk 1 above)
Comment (bsipocz) [reply {{ cookiecutter.package_name }}/.travis.yml]: This is a good point, currently in most packages we build the egg_info for all supported versions. Should we then drop all but one for python2 and one for python3?

File: {{ cookiecutter.package_name }}/.travis.yml (hunk 2 above)
Comment (pllim) [reply {{ cookiecutter.package_name }}/.travis.yml]: In Python release notes, they advised the more "conservative" users to wait till 3.7.1 for bugs to get sorted out. But that said, anyone creating a *new* package from this template might want 3.7 anyway? I'll leave that to your call.

Comment (bsipocz) [reply {{ cookiecutter.package_name }}/.travis.yml]: conservative users don't use the package template :trollface: 

I don't have strong preference either way, so I'm leaving this PR to be merged by @astrofrog.

Comment (pllim) [reply {{ cookiecutter.package_name }}/.travis.yml]: OK I don't have preference either way, so just lemme know if you want me to change it to 3.7 as global default. :smile:

**Comment by astrofrog** [comment]
I agree with @bsipocz that it probably makes sense to change the default to 3.7 (although we should make sure we have a 3.5 and 3.6 job)

**Comment by pllim** [comment]
OK, done. How about now?

**Comment by bsipocz** [comment]
Well, given the agreement and changes, I'm going ahead and merge this. Thanks @pllim!

**Pull request merged**